scripts/setupvars/setupvars.sh

```diff
@@ -72,10 +72,17 @@ if [ -z "$python_version" ]; then
     python_version=$(python3 -c 'import sys; print(str(sys.version_info[0])+"."+str(sys.version_info[1]))')
 fi
 
-version_arr=(${python_version//./ })
+# splitting Python version variable depending on the used shell 
+if [ -n "$ZSH_VERSION" ]; then
+    version_arr=(${(@s:.:)python_version})
+else
+    version_arr=(${python_version//./ })
+fi
+
 if [ "${#version_arr[@]}" -ge "2" ]; then
-    python_version_major=${version_arr[0]}
-    python_version_minor=${version_arr[1]}
+    # Use negative indices because zsh starts indexing from 1 instead of 0
+    python_version_major=${version_arr[-2]}
+    python_version_minor=${version_arr[-1]}
```

Comment:
What if array length is greater than 2?

Reply:
Good catch, I thought that `python_version` is only created in the line 72 (from python script) but I see that also user can pass python version to the script directly. I will fix it :)